core/arch/arm/mm/core_mmu.c

```diff
@@ -284,8 +284,15 @@ static size_t discovered_nsec_ddr_nelems;
 
 static int cmp_pmem_by_addr(const void *a, const void *b)
 {
-	return ((const struct core_mmu_phys_mem *)a)->addr -
-	       ((const struct core_mmu_phys_mem *)b)->addr;
+	const struct core_mmu_phys_mem *pmem_a = a;
+	const struct core_mmu_phys_mem *pmem_b = b;
+
+	if (pmem_a->va > pmem_b->va)
+		return 1;
```

Comment:
Thanks again @MrVan, this was a lazy copy/paste of mine and compiler could have not caught the bug !